### DISCUSSION TITLE
Disk full on Proxy

### DISCUSSION THREAD

**Discussion by mickeypash** [gravitational/teleport#7072] [Q&A]
Hi there,

Teleport server version: 4.4.8
Config: HA (auth + proxy)

Here's the original error when I tried to SSH onto the host.

```bash
mkdir /var/lib/teleport/log/upload/streaming/default/multi/1418464f-ad71-4ac9-ba1b-f72711e23c74: no space left on deviceerror: ssh: could not start shell
```

I had a quick look through the issues but I haven't spent enough time understanding the underlying architecture.

- Could you advise if this issue has been fixed in a later version?
- Based on the path above can you tell me if it's todo with Session recordings or logs? - maybe point me at some of the code?

Thank you for your time!
Let me know if I can provide further details.

**Comment by webvictim** [comment]
By default Teleport records session logs on the node itself and then uploads them to the auth server when the sesssion is finished. If there's no space on the node to write the session log, the connection will fail.

I don't believe this has been fixed, as the behaviour is somewhat intentional. You may be able to work around this using `proxy-sync` or `proxy-async` recording modes - see https://goteleport.com/docs/changelog/#session-streaming-improvements

We have https://github.com/gravitational/teleport/issues/5424 which describes a similar issue, so we are aware of it.

**Reply by mickeypash** [reply]
Thank you! I might reach out via our the enterprise support channel down the line. For now I'll increase the EBS attached storage.

I'm assuming you apply some compression to session recordings?
There are cases where a developer connects to a machine, runs one command and forgets to close the session for several hours.

**Reply by mickeypash** [reply]
> By default Teleport records session logs on the node itself and then uploads them to the auth server

Do you mean something like: `Node -> upload logs via http -> Auth`
